addons/web/static/src/js/core/utils.js

```diff
@@ -278,6 +278,32 @@ var utils = {
         }
         return "";
     },
+    /**
+     * Gets dataURL (base64 data) from the given file or blob.
+     * Technically wraps FileReader.readAsDataURL in Promise.
+     *
+     * @param {Blob|File} file
+     * @returns {Promise} resolved with the dataURL, or rejected if the file is
+     *  empty or if an error occurs.
+     */
+    getDataURLFromFile: function (file) {
+        return new Promise(function (resolve, reject) {
+            if (!file) {
+                return reject();
```

Comment:
I would do
```javascript
reject();
return;
```
or in fact, I would not do this in the promise implementation since it is a check that the original function receives correct arguments. I would do:
```javascript
if (!file) {
    return Promise.reject();
}
return new Promise(...);
```
and in fact, I am not sure it should return a rejected promise in this case but rather fail directly.

Reply:
In regard to the comment on the enterprise PR, we need to decide what we want to do with this.

It seems that empty file is a "normal" use case that should be "ignored".

So we have:
- working case, we want the data
- empty file: we want nothing, which includes probably not rejecting, or rejecting in a way that is different than the other errors (such as having an empty reject?)
- actual error: we want the error message

Reply:
Done, reject on !file before the main promise.